### PR TITLE
gh-100556: Better `or` doc wording

### PR DESCRIPTION
**Pull request by ramvikrams**
gh-100556: changed the or documentation

### CODE REVIEW

**Review by AlexandroLuis — CHANGES_REQUESTED** [review]
i wonder how **and** works if there's no comparison
just 
`
if x is false:
  x 
else:
  y
`
and not 
`
if x is True and y is True:
  True
else:
  False
`

**Comment by ramvikrams** [comment]
> i wonder how **and** works if there's no comparison
> just 
> `
> if x is false:
>   x 
> else:
>   y
> `
> and not 
> `
> if x is True and y is True:
>   True
> else:
>   False
> `

Thanks will do the change

File: Doc/library/stdtypes.rst [hunk 1]
```diff
@@ -84,12 +84,12 @@ These are the Boolean operations, ordered by ascending priority:
...
 +=============+=================================+=======+
-| ``x or y``  | if *x* is false, then *y*, else | \(1)  |
-|             | *x*                             |       |
-+-------------+---------------------------------+-------+
-| ``x and y`` | if *x* is false, then *x*, else | \(2)  |
+| ``x or y``  | if *x* is True, then *x*, else  | \(1)  |
 |             | *y*                             |       |
 +-------------+---------------------------------+-------+
+| ``x and y`` | if *x* is True and *y* is True, | \(2)  |
+|             | then True, else False           |       |
```
Comment (slateny) [inline Doc/library/stdtypes.rst]: I think you misunderstood the other comment, as this now doesn't cover something like

```
>>> [] and [1]
[]
```

Comment (ramvikrams) [reply Doc/library/stdtypes.rst]: could you please tell then how should it be framed, bcoz I am not able to think of it

Comment (AlexandroLuis) [reply Doc/library/stdtypes.rst]: `if x and y, then True, else False`

it covers if it's Null, empty, False...

Comment (ramvikrams) [reply Doc/library/stdtypes.rst]: thanks i'll do it

Comment (slateny) [reply Doc/library/stdtypes.rst]: The original mention of 'true'/'false' refers to all truthy and falsey values, which include 0, None, empty lists, etc., and saying `True` excludes those as `True` refers to just the boolean value. I also don't think that it's all that helpful for a table to have an entry to say `x and y` then its explanation also say `x and y`, so I would actually recommend just sticking with the original for this `and` section.

File: Doc/library/stdtypes.rst [hunk 2]
```diff
@@ -84,8 +84,8 @@ These are the Boolean operations, ordered by ascending priority:
 +-------------+---------------------------------+-------+
 | Operation   | Result                          | Notes |
 +=============+=================================+=======+
-| ``x or y``  | if *x* is false, then *y*, else | \(1)  |
-|             | *x*                             |       |
+| ``x or y``  | if *x* is True, then *x*, else  | \(1)  |
+|             | *y*                             |       |
```
Comment (hugovk) [inline Doc/library/stdtypes.rst]: @slateny's comment on 'true'/'True' also applies here:

```suggestion
| ``x or y``  | if *x* is true, then *x*, else  | \(1)  |
|             | *y*                             |       |
```

Comment (ramvikrams) [reply Doc/library/stdtypes.rst]: Thanks

**Review by CAM-Gerlach — APPROVED** [review]
LGTM IMO; it avoids a double negative (if false, else) and the order follows more closely with that of the expression itself

**Review by erlend-aasland — APPROVED** [review]

**Comment by erlend-aasland** [comment]
Thanks for the PR @ramvikrams

**Comment by CAM-Gerlach** [comment]
Oops, forgot to fix the title...hope you fixed it in the commit message :)

**Comment by erlend-aasland** [comment]
> Oops, forgot to fix the title...hope you fixed it in the commit message :)

I did... hope you like it 😄 b40dd71241f092d90c192ebff5d58cbd7e84dc52
I'll do better next time!

**Comment by CAM-Gerlach** [comment]
Oh, that's fine then of course :) At least for me updating the title-generated commit message updates the PR title as well, but it seems that's actually GitHub Refined rather than a builtin feature (I never know these days, heh)

**Pull request merged**